txnsync/txngroups_test.go

```diff
@@ -47,23 +224,287 @@ func TestTxnGroupEncoding(t *testing.T) {
 				{
 					Txn: transactions.Transaction{
 						Type: protocol.AssetConfigTx,
+						Header: transactions.Header{
+							Sender:      basics.Address(crypto.Hash([]byte("1"))),
+							Fee:         basics.MicroAlgos{Raw: 100},
+							GenesisHash: genesisHash,
+						},
 					},
+					Sig: crypto.Signature{4},
 				},
 				{
 					Txn: transactions.Transaction{
 						Type: protocol.AssetFreezeTx,
+						Header: transactions.Header{
+							Sender:      basics.Address(crypto.Hash([]byte("1"))),
+							GenesisHash: genesisHash,
+						},
 					},
+					Sig: crypto.Signature{5},
 				},
 				{
 					Txn: transactions.Transaction{
 						Type: protocol.CompactCertTx,
+						Header: transactions.Header{
+							Sender:      basics.Address(crypto.Hash([]byte("1"))),
+							GenesisHash: genesisHash,
+						},
 					},
+					Msig: crypto.MultisigSig{Version: 1},
 				},
 			},
 		},
 	}
-	encodedGroupsBytes := encodeTransactionGroups(inTxnGroups)
-	out, err := decodeTransactionGroups(encodedGroupsBytes)
+	addGroupHashes(inTxnGroups, 6, []byte{1})
+	encodedGroupsBytes, err := encodeTransactionGroups(inTxnGroups)
 	require.NoError(t, err)
-	require.Equal(t, inTxnGroups, out)
+	out, err := decodeTransactionGroups(encodedGroupsBytes, genesisID, genesisHash)
+	require.NoError(t, err)
+	require.ElementsMatch(t, inTxnGroups, out)
+}
+
+func txnGroupsData() (txnGroups []transactions.SignedTxGroup, genesisID string, genesisHash crypto.Digest, err error) {
+	dat, err := ioutil.ReadFile("../test/testdata/mainnettransactions")
```

Comment:
Looks to me like this a file that contains all the blocks rather than the transactions.
While this is fine, please note that large portion of the data is going to be the certificate, which you couldn't care less about ( and consumes large portion of this binary file size ).